man/man1/mmscene.1
.\" Automatically generated by Pandoc 2.19.2
.\"
.\" Define V font for inline verbatim, using C font in formats
.\" that render this, and otherwise B font.
.ie "\f[CB]x\f[]"x" \{\
. ftr V B
. ftr VI BI
. ftr VB B
. ftr VBI BI
.\}
.el \{\
. ftr V CR
. ftr VI CI
. ftr VB CB
. ftr VBI CBI
.\}
.TH "MMSCENE" "1" "December 07, 2021" "mmscene 2.6" "User Manual"
.hy
.SH NAME
.PP
\f[B]mmscene\f[R] - control the scenes displayed by the MMM-Scenes
MagicMirror module
.SH SYNOPSIS
.PP
\f[B]mmscene\f[R] [active|info|next|prev|name|number]
.SH DESCRIPTION
.PP
The \f[I]mmscene\f[R] command can be used to control the scenes
displayed by the MMM-Scenes MagicMirror module.
The command provided command line controls to move to the next scene,
previous scene, named scene, scene number, and can be used to retrieve
info on the active and configured scenes.
.SH COMMAND LINE OPTIONS
.TP
\f[B]mmscene next\f[R]
moves the display to the next scene
.TP
\f[B]mmscene prev\f[R]
moves the display to the previous scene
.TP
\f[B]mmscene name\f[R]
moves the display to the named scene
.TP
\f[B]mmscene number\f[R]
moves the display to scene number \[aq]number\[aq]
.TP
\f[B]mmscene active\f[R]
retrieves information on the active modules
.TP
\f[B]mmscene info\f[R]
retrieves information on the configured scenes
.SH AUTHORS
.PP
Written by Ronald Record <<github@ronrecord.com>>
.SH LICENSING
.PP
MMSCENE is distributed under an Open Source license.
See the file LICENSE in the MMSCENE source distribution for information
on terms & conditions for accessing and otherwise using MMSCENE and for
a DISCLAIMER OF ALL WARRANTIES.
.SH BUGS
.PP
Submit bug reports online at:
<<https://github.com/doctorfree/MirrorCommand/issues>>
.SH SEE ALSO
.PP
Full documentation and sources at:
<<https://github.com/doctorfree/MirrorCommand>>


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file.

18 June 2024, MirrorCommand version 4.0.1 release 2:

  Version 4.0.1 Release 2 of MirrorCommand removes the Gateway module.
  Move TelegramBot API key to module folder.
  Move release assets from Gitlab to Github.

01 January 2023, MirrorCommand version 4.0.1 release 1:

  Version 4.0.1 Release 1 of MirrorCommand replaces the use of
  module MMM-DarkSkyForecast with module MMM-OpenWeatherForecast.
  Apple bought DarkSky and is removing free access to the API.

14 April 2022, MirrorCommand version 4.0.0 release 3:

  Version 4.0.0 Release 3 of MirrorCommand adds the capability of updating
  the MagicMirror installation with the `mirror` command. The `mirror`
  command can also be used to update all installed MagicMirror modules.
  Bug fixes and improved user interface are included in this release.

20 March 2022, MirrorCommand version 4.0.0 release 2:

  Minor release update with changes to `set_asound_conf` to update the
  default card number in `/usr/local/bin/vol`, add missing config files,
  and reflect the name changes of `/usr/local/bin/reboot` and
  `/usr/local/bin/shutdown` to `/usr/local/bin/myreboot` and
  `/usr/local/bin/myshutdown`. New screenshots in README.

18 March 2022, MirrorCommand version 4.0.0 release 1:

  Version 4.0.0 of MirrorCommand includes modifications to support the
  MMM-GoogleAssistant v4 module. In addition, the MMM-Detector module has
  been replaced by the EXT-Detector module. The EXT-GooglePhotos and Gateway
  modules have been added and no longer used modules removed. Configuration
  changes necessary to support MMM-GoogleAssistan v4 and these new extension
  modules have been applied to all relevant config files. All config files are
  retained in their original non-customized format in `/usr/local/MirrorCommand/...`
  while those copied into `/usr/local/MagicMirror/...` are customized. In this way,
  subsequent re-customization can be supported.

03 February 2022, MirrorCommand version 3.0.2 release 2:

  This minor release update of MirrorCommand incorporates changes to improve
  integration with the MirrorImages package downloads. In addition, support
  is added for models by country in the MirrorImages downloads and in the
  MirrorCommand default config files. New Google Drive file ids have been
  used in the download scripts to reflect the MirrorImages distribution changes.

29 January 2022, MirrorCommand version 3.0.2 release 1:

  This release update of MirrorCommand incorporates changes to integrate with the
  MirrorImages RPM installations. The scripts that download image archives for the
  [MirrorImages](https://gitlab.com/doctorfree/MirrorImages) packages have been
  updated to execute in non-interactive mode when being run during an RPM installation.

27 January 2022, MirrorCommand version 3.0.1 release 3:

  This minor release update of MirrorCommand fixes a bug in the install
  scripts introduced by recent changes to Bash wrt the return status of
  the 'type' builtin. MMM-stocks module is now pulled from my fork to
  provide dependency on 'request' and security fixes.

22 January 2022, MirrorCommand version 3.0.1 release 2:

  This minor release update of MirrorCommand enhances support for
  RPM based Linux installation and fixes an error introduced into
  the MMM-BackgroundSlideshow module in recent commits. It is
  recommended that all RPM based systems utilize this release
  or later and any systems utilizing the MMM-BackgroundSlideshow
  module delete that module and reinstall with the `module_update`
  command included in this release.

17 January 2022, MirrorCommand version 3.0.1 release 1:

  This minor release update of MirrorCommand introduces support for
  non-interactive RPM installation.

16 January 2022, MirrorCommand version 3.0.0 release 2:

  This minor release update of MirrorCommand introduces support for
  Ubuntu 21.10 along with improved support for npm and nodejs installation.
  Image download utilities were updated to support downloads using the
  MirrorImages version 3.0.0 packages.

12 January 2022, MirrorCommand version 3.0.0 release 1:

  This major release of MirrorCommand introduces support for multiple
  monitors, switching MagicMirror screen display between monitors,
  full support for both portrait and landscape mode displays,
  and renames the project from MirrorCommandLine to MirrorCommand.
  Voice commands for screen switching have been added via the
  MMM-GoogleAssistant module recipe file MirrorCommand.js.

  Changes to packaging include the creation and distribution of an RPM
  format installation package as well as the implementation of release
  creation during the Gitlab continuous integration process.

21 December 2021, MirrorCommandLine version 2.7 release 2:

  This minor release update includes changes to enable automated
  ALSA USB audio device configuration, startup screen rotation for
  both portrait and landscape mode displays, and custom arp scan
  device lists for the MMM-MacAddressScan module.

17 December 2021, MirrorCommandLine version 2.7 release 1:

  This release of MirrorCommandLine primarily introduces support for
  landscape mode displays. All MagicMirror config files use templates
  for their display resolution settings. These and the MagicMirror CSS
  settings are set during installation of the package after detecting
  the display mode.

  Many other improvements are included in this release including fully
  automated installation and configuration. Supported systems are not
  limited to the Raspberry Pi but now include most standard Linux systems.

29 November 2021, MirrorCommandLine version 2.6 release 1:

  In this release the installed directory moved from
  /usr/local/MagicMirror to /usr/local/MirrorCommandLine

20 November 2021, MirrorCommandLine version 2.5:

  This release adds support for those not using PM2 and a shadow set of

mmscene.sh
#!/bin/bash
#
# mmscene - manage MMM-Scenes MagicMirror module scenes
#
# Author: Ronald Joe Record <ronaldrecord@gmail.com>
#
# Usage: mmscene [next|prev|name|number|info]
#   Where 'mmscene next' moves the display to the next scene
#         'mmscene prev' moves the display to the previous scene
#         'mmscene name' moves the display to the named scene
#         'mmscene number' moves the display to scene number 'number'
#         'mmscene active' retrieves information on the active modules
#         'mmscene current' retrieves information on the current scene (TBD)
#         'mmscene info' retrieves information on the configured scenes
#
# The IP address of the MagicMirror
IP="10.0.1.85"
# The port on which it is listening
PORT="8080"
# The module name
MODULE_NAME="MMM-Scenes"
# The MMM-Remote-Control API key
apikey="d8d8947ad3da477288bcbe3c7353523b"
#
modurl="http://${IP}:${PORT}/api/module/${MODULE_NAME}"
apiurl="http://${IP}:${PORT}/api/notification"
usejq=`type -p jq`

usage() {
    # printf "\nUsage: mmscene [active|current|info|next|prev|name|number]"
    printf "\nUsage: mmscene [active|info|next|prev|name|number]"
    printf "\nWhere:\n\t'mmscene next' moves the display to the next scene"
    printf "\n\t'mmscene prev' moves the display to the previous scene"
    printf "\n\t'mmscene name' moves the display to the named scene"
    printf "\n\t'mmscene number' moves the display to scene number 'number'"
    printf "\n\t'mmscene active' retrieves information on the active modules"
    # printf "\n\t'mmscene current' retrieves information on the current scene"
    printf "\n\t'mmscene info' retrieves information on the configured scenes\n"
    exit 1
}

[ "$1" ] || {
    echo "Argument required. Exiting."
    usage
}

command="$1"

case $command in
    active)
        curl -X GET http://${IP}:${PORT}/api/module?apiKey=${apikey} 2> /dev/null | jq .
        ;;
    current)
        if [ "$usejq" ]
        then
            # curl -X GET ${apiurl}/SCENES_CURRENT?apiKey=${apikey} 2> /dev/null | jq '.[].name'
            # curl -X GET "${apiurl}/SCENES_CURRENT?apiKey=${apikey}&callback=info" 2> /dev/null | jq '.payload.callback'
            echo "'mmscene current' not yet implemented"
        else
            # curl -X GET ${apiurl}/SCENES_CURRENT?apiKey=${apikey} 2> /dev/null
            echo "'mmscene current' not yet implemented"
        fi
        ;;
    info)
        if [ "$usejq" ]
        then
            curl -X GET ${modurl}?apiKey=${apikey} 2> /dev/null | jq .
        else
            curl -X GET ${modurl}?apiKey=${apikey} 2> /dev/null
        fi
        ;;
    next)
        if [ "$usejq" ]
        then
            curl -X GET ${apiurl}/SCENES_NEXT?apiKey=${apikey} 2> /dev/null | jq .
        else
            curl -X GET ${apiurl}/SCENES_NEXT?apiKey=${apikey} 2> /dev/null
        fi
        ;;
    prev)
        if [ "$usejq" ]
        then
            curl -X GET ${apiurl}/SCENES_PREV?apiKey=${apikey} 2> /dev/null | jq .
        else
            curl -X GET ${apiurl}/SCENES_PREV?apiKey=${apikey} 2> /dev/null
        fi
        ;;
    [0-9]*)
        if [ "$usejq" ]
        then
            curl -X GET "${apiurl}/SCENES_ACT?index=$command&apiKey=${apikey}" 2> /dev/null | jq .
        else
            curl -X GET "${apiurl}/SCENES_ACT?index=$command&apiKey=${apikey}" 2> /dev/null
        fi
        ;;
    *)
        if [ "$usejq" ]
        then
            curl -X GET "${apiurl}/SCENES_ACT?name=$command&apiKey=${apikey}" 2> /dev/null | jq .
        else
            curl -X GET "${apiurl}/SCENES_ACT?name=$command&apiKey=${apikey}" 2> /dev/null
        fi
        ;;
esac
